azure-pipelines/signing.yml

```diff
@@ -69,69 +69,64 @@ jobs:
+          Write-Host "##vso[task.setvariable variable=VCPKG_FULL_VERSION;isOutput=true]$env:VCPKG_INITIAL_BASE_VERSION-$headSha"
     - task: Powershell@2
       displayName: 'Lock Installer Scripts Versions'
       inputs:
         pwsh: true
         filePath: vcpkg-init/lock-versions.ps1
         arguments: '-Destination "$(Build.BinariesDirectory)" -VcpkgBaseVersion $(VCPKG_INITIAL_BASE_VERSION)'
-    # Build and test vcpkg-artifacts
     - task: UseNode@1
       displayName: Use Node 18 or later
       inputs:
         version: "18.x"
-    - script: copy azure-pipelines\.npmrc-internal ce\common\config\rush\.npmrc
-      displayName: Use internal .npmrc
-    - task: npmAuthenticate@0
-      inputs:
-        workingFile: 'ce\common\config\rush\.npmrc'
-      # The working directory change is to get the above .npmrc used when installing rush
-    - script: npm install -g @microsoft/rush
-      displayName: Install Rush
-      workingDirectory: ce\common\config\rush
-    - script: |
-        set PATH=%PATH%;%APPDATA%\npm
-        rush update
-      displayName: Install vcpkg-artifacts Dependencies
-      workingDirectory: ce
-    - script: |
-        set PATH=%PATH%;%APPDATA%\npm
-        rush lint
-      displayName: Check vcpkg-artifacts for Linting Errors
-      workingDirectory: ce
+    - task: Npm@1
+      inputs:
+        command: 'custom'
+        workingDir: 'vcpkg-artifacts'
+        customCommand: 'install'
+        customRegistry: 'useFeed'
+        customFeed: '0bdbc590-a062-4c3f-b0f6-9383f67865ee/105b4584-173c-41aa-8061-612294abe099'
+      displayName: Install vcpkg-artifacts Dev Dependencies
+    - task: ComponentGovernanceComponentDetection@0
+      displayName: Detect Components
+      inputs:
+        sourceScanPath: vcpkg-artifacts
     - task: CodeQL3000Init@0
-      displayName: 'CodeQL Initialize'
+      displayName: CodeQL Initialize
     - script: |
-        set PATH=%PATH%;%APPDATA%\npm
-        rush rebuild
-      displayName: Build vcpkg-artifacts Packages
-      workingDirectory: ce
+        mkdir "$(Build.BinariesDirectory)"
+        mkdir "$(Build.BinariesDirectory)\vcpkg-artifacts"
+        node "$(Build.SourcesDirectory)\vcpkg-artifacts\node_modules\typescript\bin\tsc" -p "$(Build.SourcesDirectory)\vcpkg-artifacts" --outDir "$(Build.BinariesDirectory)\vcpkg-artifacts"
+      displayName: Build TypeScript
     - task: CodeQL3000Finalize@0
-      displayName: 'CodeQL Finalize'
+      displayName: CodeQL Finalize
+    - task: Npm@1
+      inputs:
+        command: 'custom'
+        workingDir: 'vcpkg-artifacts'
+        customCommand: 'install --omit=dev'
+        customRegistry: 'useFeed'
+        customFeed: '0bdbc590-a062-4c3f-b0f6-9383f67865ee/105b4584-173c-41aa-8061-612294abe099'
+      displayName: Install vcpkg-artifacts Prod Dependencies
+    - script: | # This script must be kept in sync with vcpkg-artifacts-target in CMakeLists.txt
+        rmdir /s /q "$(Build.BinariesDirectory)\vcpkg-artifacts\test"

```

Comment:
I'm torn on it. On the one hand, I don't like having the steps duplicated. On the other, because this needs to tear the results apart, sign them, and glue them back together again, I really don't know how to represent that as any CMake analogue. (And we currently don't run a cmake configure on this node)